numpy/random/bit_generator.pyx

```diff
@@ -0,0 +1,499 @@
+        return x.copy()
+    elif isinstance(x, str):
+        if x.startswith('0x'):
+            x = int(x, base=16)
+        elif DECIMAL_RE.match(x):
+            x = int(x)
+        else:
+            raise ValueError("unrecognized seed string")
+    if isinstance(x, (int, np.integer)):
+        return _int_to_uint32_array(x)
+    elif isinstance(x, (float, np.inexact)):
+        raise TypeError('seed must be integer')
+    else:
+        if len(x) == 0:
+            return np.array([], dtype=np.uint32)
+        # Should be a sequence of interpretable-as-ints. Convert each one to
+        # a uint32 array and concatenate.
+        subseqs = [_coerce_to_uint32_array(v) for v in x]
+        return np.concatenate(subseqs)
+
+
+cdef uint32_t hashmix(uint32_t value, uint32_t * hash_const):
+    # We are modifying the multiplier as we go along, so it is input-output
+    value ^= hash_const[0]
+    hash_const[0] *= MULT_A
+    value *=  hash_const[0]
+    value ^= value >> XSHIFT
+    return value
+
+cdef uint32_t mix(uint32_t x, uint32_t y):
+    cdef uint32_t result = (MIX_MULT_L * x - MIX_MULT_R * y)
+    result ^= result >> XSHIFT
+    return result
+
+
+cdef class SeedSequence():
+    """
+    SeedSequence(entropy=None, program_entropy=None, spawn_key=(), pool_size=%d)
+    creates an appropriate seed sequence for BitGenerators via the
+    `generate_state` method. Calling `spawn(n) <spawn>` will create n
+    SeedSequences that can be used to seed independent BitGenerators, i.e. for
+    different threads.
+
+    Parameters
+    ----------
+    entropy: {None, int, sequence[int]}, optional
+        The entropy for creating a SeedSequence.
+
+    program_entropy: {None, int, sequence[int]}, optional
+        A second source of entropy, typically per-application
+
+    spawn_key: {(), sequence[int]}, optional
+        A third source of entropy, used internally when calling
+        `SeedSequence.spawn`
+
+    """ % DEFAULT_POOL_SIZE
+
+    def __init__(self, entropy=None, program_entropy=None, spawn_key=(),
+                 pool_size=DEFAULT_POOL_SIZE):
+        if pool_size < DEFAULT_POOL_SIZE:
+            raise ValueError("The size of the entropy pool should be at least "
+                             f"{DEFAULT_POOL_SIZE}")
+        if entropy is None:
+            entropy = randbits(pool_size * 32)
+        elif not isinstance(entropy, (int, np.integer, list, tuple, range,
+                                      np.ndarray)):
+            raise TypeError('SeedSequence expects int or sequence of ints for '
+                            'entropy not {}'.format(entropy))
+        self.entropy = entropy
+        self.program_entropy = program_entropy
+        self.spawn_key = tuple(spawn_key)
+        self.pool_size = pool_size
+        self.n_children_spawned = 0
+
+        self.pool = np.zeros(pool_size, dtype=np.uint32)
+        self.mix_entropy(self.pool, self.get_assembled_entropy())
+
+    def __repr__(self):
```

Comment:
While a repr is useful for humans, it would also be useful to know to how recreate the precise node in a SeedSequence programmatically, which would require something like `state`  from a `BitGenerator`. Essentially repr for machines (returning a dict?)

Reply:
added `state` property and test that it can be used `SeedSequence(**s.state)`

Reply:
`n_children_spawned` is also required.